core/modules/wdb/ChunkResource.cc

```diff
@@ -196,9 +206,38 @@ class Backend {
         }
         return true;
     }
+
     void discard(ScTableVector const& v) {
         _discard(v.begin(), v.end());
     }
+
+    enum LockStatus {UNLOCKED, LOCKED_OTHER, LOCKED_OURS};
+    std::string strLockStatus(LockStatus ls) {
+        std::string s = "unknown";
+        switch (ls) {
+        case UNLOCKED: s = "UNLOCKED";
+            break;
+        case LOCKED_OTHER: s = "LOCKED_OTHER";
+            break;
+        case LOCKED_OURS: s = "LOCKED_OURS";
+            break;
+        }
+        return s;
+    }
+
+    bool memLockCheck() {
+        if (_isFake) {
+            return true;
+        }
+        LockStatus mls = _memLockStatus();
+        if (mls != LOCKED_OURS) {
+            LOGF_ERROR("memLockCheck could not verify this program owned the memory table lock, Exiting.");
+            exit(EXIT_FAILURE);
+            return false;
+        }
+        return true;
+    }
```

Comment:
I don't think this function's return value is useful, and the name could be clearer. Would something like:
```
void requireMemLockOwnership() {
    if (!_isFake && _memLockStatus() != LOCKED_OURS) {
        LOGF_ERROR(...)
        exit(...);
    }
}
```
be better?